packages/schema-type/src/solidity/SchemaType.sol

```diff
@@ -0,0 +1,236 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+using { getStaticByteLength } for SchemaType global;
+
+// WARNING: SchemaType methods use hardcoded enum indexes, review them after any changes to the enum
```

Comment:
Should we add the same comment in the ts file? (And maybe a comment about always changing all implementations, never just an individual one?)

Reply:
Added more warnings. I think solidity file gets the core warning to mirror things everywhere, and the other langs should just copy the solidity file